### DISCUSSION TITLE
Automatically suppress linter errors

### DISCUSSION THREAD

**Discussion by allanlewis** [biomejs/biome#3949] [Ideas]
As per [my Discord thread](https://discord.com/channels/1132231889290285117/1285240960690688114), I'd love to see something similar to [Ruff's `--add-noqa` option](https://docs.astral.sh/ruff/linter/#inserting-necessary-suppression-comments), which adds linter-suppressing comments for each error.

My standard workflow when adding Ruff to an existing Python codebase is to run:
```shell
ruff check --fix
ruff check --add-noqa
ruff check
```
This fixes all safely-fixable errors, suppresses the rest, and then checks that there are no remaining errors. Perhaps we could enable a similar workflow in Biome?

Aside, one thing that might make this easier would be if the JSON output (`biome lint --reporter=json`) gave the line number of each error rather than the character number, then one could use `sed` or similar to insert suppression comments.

**Comment by ematipico** [comment]
Out of curiosity, what does `noqa` stand for? I don't know python and its ecosystem

**Reply by nhedger** [reply]
My guess is no quality assurance

**Reply by allanlewis** [reply]
Yes, "noqa" means "no quality assurance", it comes from one of the early Python linters and has been adopted by Ruff.

**Comment by anthonyshew** [comment]
Working on this feature here!: https://github.com/biomejs/biome/issues/4007

**Reply by eMerzh** [reply]
i was wondering if by doing that we couldn't migrate eslint comments to biome 🤔 
it's not exactly the same, but it looks like there are some common stuff

**Reply by ematipico** [reply]
That's out of scope for this proposal.

Plus, it isn't possible to map all kinds of Eslint suppression comments to Biome

**Reply by anthonyshew** [reply]
I'll also add: As I'm converting to Biome, I remove some rule in ESLint and then ESLint complains about an unused ignore comment. Running ESLint with `--fix` removes the comment. Pretty quick to get rid of the unneeded comments with ESLint itself!
